src/Nethermind/Nethermind.Synchronization/BeamSync/BeamBlockchainProcessor.cs

```diff
@@ -205,7 +205,8 @@ private void CancelAllBeamSyncTasks()
             env.BlockProcessor.TransactionProcessed += (sender, args) =>
             {
                 Interlocked.Increment(ref Metrics.BeamedTransactions);
-                if (_logger.IsInfo) _logger.Info($"Processed tx {args.Index + 1}/{block.Transactions.Length} of {block.Number}");
+                // we can safely use the null forgiving operator on block.Transaction here because this method will only get called when a transaction is processed, which means the block had at least one transaction
```

Comment:
I am thinking of adding transactions to the constructor and make the block Transactions non-nullable first